pkg/storage/segments/store.go

```diff
@@ -259,6 +268,93 @@ func (s *segmentStore) Delete(ctx context.Context, path paths.Path) (err error)
 	return s.pdb.Delete(ctx, path)
 }
 
+// Repair retrieves an at-risk segment and repairs and stores lost pieces on new nodes
+func (s *segmentStore) Repair(ctx context.Context, path paths.Path, lostPieces []int) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	//Read the segment's pointer from the PointerDB
+	pr, err := s.pdb.Get(ctx, path)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	if pr.GetType() == pb.Pointer_REMOTE {
+		seg := pr.GetRemote()
+		pid := client.PieceID(seg.PieceId)
```

Comment:
you can just do GetRemote().GetPieceId()

Reply:
was giving me "cannot use pid (type string) as type client.PieceID in argument to s.ec.Get" error. so changed it back. Also in segment get, it is used as above so used same function call